packages/devops/scripts/deployment/pushE2eDumpLogToS3.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+# Run directly from a nightly cron job, so needs full setup of path etc.
+
+# Set PATH to include depencencies
+export PATH=/home/ubuntu/.local/bin:/home/ubuntu/.yarn/bin:/home/ubuntu/.config/yarn/global/node_modules/.bin:/home/ubuntu/.nvm/versions/node/v12.18.3/bin:/usr/local/bin:$PATH
+
+DIR=$(dirname "$0")
+export STAGE=$(${DIR}/../utility/getEC2TagValue.sh Stage)
+INSTANCE_NAME=$(${DIR}/../utility/getEC2TagValue.sh Name)
+
+if [[ $1 == "--force" ]]; then
+    force=true
+fi
+
+local_log="/home/ubuntu/logs/dump_e2e_data.log"
+s3_folder_uri="s3://tupaia/dumps/e2e"
+
+if [[ "$STAGE" == "e2e" || $force == true ]]; then
+    s3_log=$(sed -En "s|.*$s3_folder_uri/(.*).tgz|\1.log|p" $local_log)
```

Comment:
I can't parse this so am just trusting it!

Reply:
Haha let me try to explain then - I'll tell you this line was more complicated on my first iteration:

* `-E` enables extended regex - that way we avoid escaping `()` in the regex
* Did you know you can use separators other than `/` in `sed`? I didn't! It's very convenient here, since we avoid escaping all the `/` in `$s3_folder_uri`. I used `|` as a separator, which is an invalid path char
* `\1` is the first matching group in the regex

So in summary: find a line which includes `$s3_folder_uri/` => `s3://tupaia/dumps/e2e/`, then grab everything after it and until `.tgz` - this will be the base name of the generated file. Finally, append `.log` to it. Eg
Input:  `(...other stuff) e2e-test-fixes_1618393495_dumps.tgz`
Output:  `e2e-test-fixes_1618393495_dumps.log`

Reply:
Nice!